package.json

```diff
@@ -18,13 +20,22 @@
     "url": "https://github.com/guardian/types/issues"
   },
   "homepage": "https://github.com/guardian/types#readme",
-  "dependencies": {
-    "typescript": "^3.8.3"
-  },
+  "dependencies": {},
```

Comment:
Good point, we're building to declaration files now so TS is optional!

One thing occurred to me: presumably declaration files are still going to rely on certain TS features that may not be available in all versions? Perhaps we could specify a version of TS in `package.json`'s [`optionalDependencies`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#optionaldependencies) for Guardian packages?

Reply:
yeah I think 3.8 should be peerdep here?

Reply:
I thought about that, I'm not sure. Here are [the Yarn docs](https://classic.yarnpkg.com/en/docs/dependency-types/#toc-peerdependencies) on the topic (I know this is npm but the principle is the same). Is this like a React/ReactDOM situation, and therefore a peer dep, or is it a "not everyone will have this installed, but that's fine"?

I think peer deps will cause npm/yarn warnings if they're not available, which doesn't feel right if we're accepting that it's valid to install this library even if you're not using TypeScript?

Reply:
what would you install this for if not for ts types?

Reply:
Technically I think you could use `Option` and `Result` without TypeScript, although I don't know if you'd want to. But yeah fair point, for this repo perhaps it's not that important.

However I was thinking more generally for our `@guardian` packages - it might make sense for e.g. `image-rendering` and `atoms-rendering` to say "hey, you can use this package with or without TypeScript, but for those who are working with TypeScript we're going to specify the version you need in `optionalDependencies` so that you don't hit problems with our declaration files".